packages/unigraph-dev-backend/src/utils/entityUtils.ts

```diff
@@ -19,9 +24,39 @@ function buildUnigraphEntityPart (rawPart: any, makeAbstract: boolean = false):
     return {"_value": unigraphPartValue};
 }
 
-export function buildUnigraphEntity (raw: Object): EntityDgraph<string> {
-    return {
-        "dgraph.type": "Entity",
-        "_value": buildUnigraphEntityPart(raw)
-    };
+export function validateEntityWithSchema(object: Object, schemaName: string) {
+    if (schemaName === "any") {
+        return true;
+    } else {
+        // TODO: Add schema validation
+        return true;
+    }
+}
+
+export function validateUnpaddedEntity(object: Object) {
+    // TODO: Validate unpadded entity
+    return true;
+}
+
+/**
+ * Builds a Unigraph entity with a given object and its schema.
+ * 
+ * For more information on how the data model works, refer to the `/docs/schemas_and_objects.md` documentation.
+ * 
+ * @param raw The raw, unpadded object (or padded object if `padding` is false). However, `$ref` may be used.
+ * @param schemaName The name of the schema object. If no schema is given, it would be given an `$/schema/any` schema.
+ * @param validateSchema Whether to validate schema. Defaults to false. If validation is on and the schema does not match the object, an error would be returned instead.
+ * @param padding Whether to pad the raw object, used to create/change objects with predicate-properties defined. If this is false and the object does not follow the data model, an error would be returned instead.
+ */
+export function buildUnigraphEntity (raw: Object, schemaName: string = "any", validateSchema: boolean = false, padding: boolean = true): EntityDgraph<string> | TypeError {
+    // Check for unvalidated entity
+    if (!validateEntityWithSchema(raw, schemaName) || (padding === false && !validateUnpaddedEntity(raw))) {
+        return new TypeError("Entity validation failed for entity " + raw)
+    } else {
+        return {
+            "type": makeRefUnigraphId(`$/schema/${schemaName}`) as RefUnigraphIdType<`$/schema/${string}`>,
```

Comment:
do you get a compiler error if this type assertion is left out? It seems like the return value of `makeRefUnigraphId` should set things up correctly (as far as I can tell)

Reply:
Yeah, seems like an issue with type resolution in TS and without assertion it just resolves it as `RefUnigraphIdType<string>`. Note that template string type support is still experimental (introduced in 4.11 rc I think)